tools/swift-inspect/Sources/SwiftInspectLinux/ElfFile.swift

```diff
@@ -0,0 +1,162 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+// TODO: replace this implementation with general purpose ELF parsing support
+// currently private to swift/stdlib/public/Backtrace.
+class ElfFile {
+  public enum ELFError: Error {
+    case readFailure(_ filePath: String, offset: UInt64, size: UInt64)
+    case notELF64(_ filePath: String, _ description: String = "")
+    case malformedFile(_ filePath: String, _ description: String = "")
+  }
+
+  public typealias SymbolMap = [String: (start: UInt64, end: UInt64)]
+
+  let filePath: String
+  let file: FileHandle
+  let ehdr: Elf64_Ehdr
+
+  public init(filePath: String) throws {
+    self.filePath = filePath
+
+    let file = try FileHandle(forReadingFrom: URL(fileURLWithPath: filePath))
```

Comment:
Why not use `Data(contentsOf:options:)` and mmap the file and read the first n bytes?

Reply:
I typically prefer explicit file I/O so you can handle failures, but I am happy to rewrite this class using `mmap` file if you prefer that.

Reply:
Reworked the `ElfFile` implementation to use `Data(options: .alwaysMapped` instead of file seek/read.